pkg/driver/helmdriver.go

```diff
@@ -15,26 +15,28 @@ import (
 	"helm.sh/helm/v3/pkg/cli"
 	"helm.sh/helm/v3/pkg/registry"
 	"helm.sh/helm/v3/pkg/storage/driver"
+	"k8s.io/client-go/rest"
 
 	api "github.com/aws/eks-anywhere-packages/api/v1alpha1"
 	auth "github.com/aws/eks-anywhere-packages/pkg/authenticator"
 )
 
 // helmDriver implements PackageDriver to install packages from Helm charts.
 type helmDriver struct {
-	cfg      *action.Configuration
-	log      logr.Logger
-	settings *cli.EnvSettings
+	cfg        *action.Configuration
+	secretAuth auth.Authenticator
+	log        logr.Logger
+	settings   *cli.EnvSettings
 }
 
 var _ PackageDriver = (*helmDriver)(nil)
 
-func NewHelm(log logr.Logger) (*helmDriver, error) {
+func NewHelm(log logr.Logger, k8sconfig *rest.Config) (*helmDriver, error) {
 	settings := cli.New()
 
 	// TODO Catch error here if not provided docker config or continue without an authfile if possible
-	secretAuth := auth.NewHelmSecret()
-	authfile, _ := secretAuth.AuthFilename()
+	secretAuth, _ := auth.NewECRSecret(k8sconfig)
+	authfile := secretAuth.AuthFilename()
```

Comment:
I see no reason for this to be done here. What concern is this of the helm driver's?

Reply:
Moved out of helm driver